### PR TITLE
feat(PPDSC-2313): script to run applitools locally

### PR DESCRIPTION
**Pull request by mstuartf**
[PPDSC-2313](https://nidigitalsolutions.jira.com/browse/PPDSC-2313)

**What**

1. Background - why this is needed
- Lack of information in the repo on how to run Applitools tests locally
- Pipeline debugging is slow and expensive
3. What did you do
- Shell script to set env variables (inc asking for api key)
- Add scripts to package json to run comp and doc tests
- Considered dockerising the script but felt like overkill
4. What does the reviewers should expect
- To be able to run applitools tests locally using a single command

The script will guide the user to run fewer tests by using the `include` config option, then ask for an api key:
Screenshot 2022-08-01 at 16 29 35

Then when the suite has finished the results and a link to the batch in the Applitools UI are shown:
Screenshot 2022-08-01 at 16 33 12

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2313-run-applitools-locally
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2313-run-applitools-locally/storybook

**Review by tbradbury — APPROVED** [review]

**Review by Srinivasan-Ramasamy — APPROVED** [review]

File: scripts/applitools_local.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+STATUS=$(git status)
+
+# the commit needs to be pushed (so that the Applitools / GitHub integration can retrieve its info)
+if [[ "$STATUS" != *"Your branch is up to date with 'origin/${BRANCH_NAME}'."* ]]; then
+  echo "Please make sure you are up-to-date with remote before running Applitools tests."
+  exit 1
+fi
+
+# guide the user to only running a subset of tests to keep usage down
+if [[ $1 = "comps" ]]; then
```
Comment (Vanals) [inline scripts/applitools_local.sh:17]: Out of curiosity, where $1 will come from? and what is it 🤔 ? I have seen it in other sh scripts but don't remember

**Comment by Vanals** [comment]
How this script will be fitting the development flow? In which scenario `can` be run or should be run?
As you mentioned, I guess, we can run it when we already know it is gonna fail and we don't want to wait the whole pipeline to run? Ta!

File: scripts/applitools_local.sh (hunk 1 above)
Comment (mstuartf) [reply scripts/applitools_local.sh:17]: It's the first arg passed to the script (in our case it will either be `comps` or `docs`.

Comment (Vanals) [reply scripts/applitools_local.sh:17]: ta!

**Comment by mstuartf** [comment]
> How this script will be fitting the development flow? In which scenario `can` be run or should be run? As you mentioned, I guess, we can run it when we already know it is gonna fail and we don't want to wait the whole pipeline to run? Ta!

Yep. The idea is that if you know you will have to resolve some tests because you have made visual changes, you don't have to run the whole pipeline (and all >1,600 tests) before being able to resolve them. You can run them locally, resolve any changes, then when your pipeline runs they should all pass. 

Alternatively, if you want to make sure that you have _not_ made any visual changes that are going to cause the pipeline to fail, this script is a quick way to check. 

The general idea is that this will make resolving and debugging Applitools quicker for devs and will reduce our usage.

**Pull request merged**